stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/NonMemberChatEventHandler.kt

```diff
@@ -0,0 +1,139 @@
+package io.getstream.chat.android.offline.querychannels
+
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.api.models.FilterObject
+import io.getstream.chat.android.client.api.models.QueryChannelsRequest
+import io.getstream.chat.android.client.call.await
+import io.getstream.chat.android.client.events.ChannelUpdatedByUserEvent
+import io.getstream.chat.android.client.events.ChannelUpdatedEvent
+import io.getstream.chat.android.client.events.MemberAddedEvent
+import io.getstream.chat.android.client.events.MemberRemovedEvent
+import io.getstream.chat.android.client.events.NotificationAddedToChannelEvent
+import io.getstream.chat.android.client.events.NotificationMessageNewEvent
+import io.getstream.chat.android.client.events.NotificationRemovedFromChannelEvent
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Filters
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.client.utils.map
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.runBlocking
+
+/**
+ * Default implementation of [ChatEventHandler] which is more generic than [MessagingChatEventHandler]. It skips updates
+ * and makes an API request if a channel wasn't yet handled before when receives [NotificationAddedToChannelEvent],
+ * [NotificationMessageNewEvent], [NotificationRemovedFromChannelEvent].
+ */
+public class NonMemberChatEventHandler(
```

Comment:
Why do you call it `NoMember...` if for member events it makes request?

Reply:
I updated the documentation now. This is the ChatEventHanlder to be used when filter is asking for channels which the current user is not a member.